src/generators.cpp

```diff
@@ -364,8 +324,8 @@ void Generator::AppendTokens(cpu_span<const int32_t> input_ids) {
     throw std::runtime_error("input_ids is empty");
   if ((input_ids.size() / state_->params_->search.batch_size) + search_->GetSequenceLength() > state_->params_->search.max_length)
     throw std::runtime_error("input_ids size (" + std::to_string(input_ids.size()) + ") + current sequence length (" + std::to_string(search_->GetSequenceLength()) + ") exceeds max length (" + std::to_string(state_->params_->search.max_length) + ")");
-  if (model_->config_->model.type == "whisper" || model_->config_->model.type == "phi3v")
-    throw std::runtime_error("Please use params.SetInputs for " + model_->config_->model.type + ". AppendTokens is not supported for this model type.");
+  if (!ModelType::IsLLM(model_->config_->model.type) && !ModelType::IsALM(model_->config_->model.type) && !ModelType::IsPipe(model_->config_->model.type))
```

Comment:
Should we check for inclusion instead of exclusion?

Only check for MultiModalModels? Since whisper is a audio language model, would that make this function not available for whisper? It was previously marked as being valid for whisper.